python/pyspark/sql/functions.py

```diff
@@ -1586,15 +1651,25 @@ def when(condition, value):
     >>> df.select(when(df.age == 2, df.age + 1).alias("age")).collect()
     [Row(age=3), Row(age=None)]
     """
-    sc = SparkContext._active_spark_context
+    sc = SparkContext._active_spark_context  # type: ignore[attr-defined]
     if not isinstance(condition, Column):
         raise TypeError("condition should be a Column")
     v = value._jc if isinstance(value, Column) else value
     jc = sc._jvm.functions.when(condition._jc, v)
     return Column(jc)
 
 
-def log(arg1, arg2=None):
+@overload   # type: ignore[no-redef]
```

Comment:
Do we need the `type: ignore[no-redef]` annotation here?
What's the error like without it?

Reply:
I am afraid so, otherwise we will hit
```
python/pyspark/sql/functions.py:1673: error: Name "log" already defined on line 479
Found 1 error in 1 file (checked 315 source files)
```

Reply:
So there are two `log` function here?
I guess we should remove one of them? cc @HyukjinKwon

Reply:
yy